### PR TITLE
Add support for ping events

### PR DESCRIPTION
**Pull request by johnworth**
It's dewey's turn to have support for ping requests added.

There's a bit of code rearranging that I needed to do, but I tried to keep the changes minimal.

There's a companion pull request for the configuration changes at https://github.com/cyverse-de/ansible/pull/4.

### CODE REVIEW

File: src/dewey/amq.clj [hunk 1]
```diff
@@ -68,12 +68,8 @@
    Throws:
      It will throw an exception if it fails to connect to the AMQP broker, setup the exchange, or
      setup the queue."
-  [host port user password queue-name exchange-name exchange-durable exchange-auto-delete qos consumer-fn & topics]
-  (consume (rmq/connect {:host                  host
-                         :port                  port
-                         :username              user
-                         :password              password
-                         :automatically-recover true})
+  [uri queue-name exchange-name exchange-durable exchange-auto-delete qos consumer-fn & topics]
```
Comment (strootman) [inline src/dewey/amq.clj:68]: The documentation for this needs to be updated.

File: src/dewey/core.clj [hunk 2]
```diff
@@ -46,32 +47,58 @@
...
+    (catch Exception e
+      (log/info e "Failed to attach to the AMQP broker. Retrying...")
+      false)))
+
+
+(defn- connect-to-events-broker?
+  []
+  (try
+    (amq/attach-to-exchange (cfg/amqp-events-uri)
+                            ""
```
Comment (strootman) [inline src/dewey/core.clj]: Specifying a non-default queue name could be helpful for debugging/administration.

File: src/dewey/core.clj [hunk 3]
```diff
@@ -46,32 +47,58 @@
...
+                            (cfg/amqp-events-exchange)
+                            (cfg/amqp-events-exchange-durable?)
+                            (cfg/amqp-events-exchange-auto-delete?)
+                            (cfg/amqp-qos)
+                            events/consume-msg
+                            "events.dewey.#")
+    (log/info "Attached to the events AMQP broker.")
```
Comment (strootman) [inline src/dewey/core.clj]: When you setup the queue name, you can emit the name in this log message.

**Review by strootman — CHANGES_REQUESTED** [review]

File: src/dewey/curation.clj [hunk 4]
```diff
@@ -313,7 +313,7 @@
```
Comment (strootman) [inline src/dewey/curation.clj:313]: I'm thinking about how to go about documenting our events not at the service level, but at the architecture level.
The `dewey.curation.resolve-consumer` function has the list of routing keys used by `dewey`, just as the `info-typer.messaging.routing-functions` function contains routing keys used by `info-typer`.

For now, this is fine. But, from my experience, tracking this stuff by diving into each service can be quite laborious. We should consider ways of centralizing the routing keys in a way that makes them "traceable", somehow.

File: src/dewey/events.clj [hunk 5]
```diff
@@ -0,0 +1,26 @@
+(ns dewey.events
+  (:require [clojure.tools.logging :as log]
+            [langohr.basic :as lb]
+            [dewey.config :as cfg])
+  (:import [org.cyverse.events.ping PingMessages
+                                    PingMessages$Ping
+                                    PingMessages$Pong]
+           [com.google.protobuf.util JsonFormat]))
+
+(defn- ping-handler
```
Comment (strootman) [inline src/dewey/events.clj:10]: This handler is nearly identical to the one in `info-typer` at some point in the future, we should consolidate this functionality. For now, it's fine.

**Review by strootman — COMMENTED** [review]
These are just comments. No change requested.

File: src/dewey/amq.clj (hunk 1 above)
Comment (johnworth) [reply src/dewey/amq.clj:68]: Done.

File: src/dewey/core.clj (hunk 2 above)
Comment (johnworth) [reply src/dewey/core.clj]: Done.

File: src/dewey/core.clj (hunk 3 above)
Comment (johnworth) [reply src/dewey/core.clj]: Done.

**Review by strootman — APPROVED** [review]

**Pull request merged**